sdk/storage/AzureStorageBlob/Source/Data/Transfer.swift

```diff
@@ -42,6 +42,9 @@ public struct TransferProgress {
 
     /// Percentage of the transfer that is complete, as a Float between 0 and 1.
     public var asFloat: Float {
+        // Returning NaN sometimes results in progress appearing as 100% when it shouldn't.
+        // This ensures progress will report 0%.
+        if totalBytes <= 0 { return 0 }
```

Comment:
Should we also handle the situation where bytes > totalBytes (IDK what happened here, maybe just return 1.0?) and bytes == totalBytes (explicitly return 1.0 here to avoid potential float division weirdness?)

Reply:
Tested for the 1.0 case and included an assertion for the abnormal case so it will crash during development but not production. In production it would simply return 1.x.